content/libraries/rio.md

```diff
@@ -364,12 +364,13 @@ sayGoodbye = do
 ```
 
 There's a bit of a problem here though. The `sayTime` action requires
-an `App` environment. However, it never actually uses the `appName`
-field. This can make it harder to use in a situation where there is no
-actual name. Also, if we wanted to define this in a library to be used
-in multiple projects, we'd have to hard-code this one specific
-application type, which wouldn't apply to many applications. (This
-also applies to our `say` function.) We need something more flexible.
+an `App` environment. However, it never actually uses any of the field
```

Comment:
This is an inaccurate change. The `sayTime` action _does_ use the other fields, just not directly.

Reply:
Yeah, I missed that. The `say` function uses the machinery. I will revert this back. There is another sentence here which confuses me:

> This can make it harder to use in a situation where there is no actual name.

What do you mean by "there is no actual name" ?

Reply:
I meant that there is no name to fill in, and therefore we don't know how to construct a value which requires a name field